tests/functional/test_newsletter_embed.py

```diff
@@ -0,0 +1,75 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+from selenium.common.exceptions import TimeoutException
+
+from pages.home import HomePage
+from pages.about import AboutPage
+from pages.contribute.contribute import ContributePage
+from pages.mission import MissionPage
+from pages.firefox.all import FirefoxAllPage
+from pages.firefox.desktop.desktop import DesktopPage
+from pages.firefox.desktop.customize import CustomizePage
+from pages.firefox.desktop.all import FirefoxDesktopBasePage
+from pages.firefox.sync import FirefoxSyncPage
+from pages.plugincheck import PluginCheckPage
+from pages.smarton.landing import SmartOnLandingPage
+from pages.smarton.base import SmartOnBasePage
+
+
+@pytest.mark.nondestructive
+@pytest.mark.parametrize(('page_object', 'kwargs'), [
+    pytest.mark.smoke((HomePage, None)),
+    (AboutPage, None),
+    pytest.mark.smoke((ContributePage, None)),
+    (MissionPage, None),
+    (FirefoxAllPage, None),
+    pytest.mark.smoke((DesktopPage, None)),
+    (CustomizePage, None),
+    (FirefoxDesktopBasePage, {'slug': 'fast'}),
+    (FirefoxDesktopBasePage, {'slug': 'trust'}),
+    (FirefoxSyncPage, None),
+    pytest.mark.skip_if_not_firefox((PluginCheckPage, None),
+        reason='https://bugzilla.mozilla.org/show_bug.cgi?id=1245208'),
+    (SmartOnLandingPage, None),
+    pytest.mark.skip_if_not_firefox((SmartOnBasePage, {'slug': 'tracking'}),
+        reason='Newsletter is only shown to Firefox browsers.'),
+    pytest.mark.skip_if_not_firefox((SmartOnBasePage, {'slug': 'security'}),
+        reason='Newsletter is only shown to Firefox browsers.'),
+    pytest.mark.skip_if_not_firefox((SmartOnBasePage, {'slug': 'surveillance'}),
+        reason='Newsletter is only shown to Firefox browsers.')])
+def test_newsletter_default_values(page_object, kwargs, base_url, selenium):
+    if kwargs:
+        page = page_object(base_url, selenium, **kwargs).open()
+    else:
+        page = page_object(base_url, selenium).open()
```

Comment:
I wonder if this would work:

```python
url_kwargs = url_kwargs or {}
page = page_class(base_url, selenium, **url_kwargs).open()
```

Slightly less duplication. Also note `url_` prefix, which I think makes it a little clearer what these keywords are used for.